authz/providers/azure/rbac/rbac.go

```diff
@@ -375,7 +378,94 @@ func (a *AccessInfo) CheckAccess(request *authzv1.SubjectAccessReviewSpec) (*aut
 
 		finalStatus = status
 	}
-	return finalStatus, nil
+	if finalStatus != nil && finalStatus.Allowed {
+		klog.V(5).Infof("Checkaccess request is allowed for user %s", checkAccessUsername)
+		return finalStatus, nil
+	}
+
+	if !a.useManagedNamespaceResourceScopeFormat || a.clusterType != managedClusters {
```

Comment:
Since we already know this is it necessary to traverse both paths?
`Logically, it means Guard will allow the request iff (#1 OR #2a OR #2b). `
Can we not do either #1 or (#2a or #2b)?

Reply:
Hmm can you elaborate on that? The 2b is not implemented yet, that's for a later step of the project

Reply:
I mean does it have to be a fallback or based on `a.useManagedNamespaceResourceScopeFormat` can we switch between 1 and 2a?